### ISSUE TITLE
Античит

### ISSUE THREAD

**Issue by Bezbasheniy** [PavelZinchenko/event-horizon-frontier-builds#213]
Участилась проблема с читерами на арене. Попадаються пользователи с невообразимым уроном или неуязвимыми кораблями в следствии теряеться почьти тысяча рейтинга а набираеться он после 96 тысяч всего по 5-19 единиц. В случае прохождения 11арены не набираеться вовсе, а потери при поражении топ 1-2 800-900 очков рейтинга. При том в топе восседает читер с умноженым уроном. Просьба ввести в проэкт нормально работающий античит.

**Comment by IsmollCheeKen** [comment]
Fun fact: the anti-cheat flags people on older build versions/game versions. You can only imagine how many people are on older build versions, so much so that even beta testers are getting branded as cheaters.

**Comment by Inedilarion** [comment]
В античит следует добавить (разумеется если этого там ещё нет):
1) Проверку характеристик модулей, если флоты переносятся на сервер не в формате сугубо внутриигровых билдов кораблей, а с учётом их характеристик (насколько я понимаю, это неизбежно после появления учёного)
2) Проверку бустов атаки/защиты базы и кораблей от общей прокачки навыков, от уровня и от капитанов

**Comment by Inedilarion** [comment]
Я встречался с читерами, видел как турель аванпоста приделали к базе и она умудрялась ремонтных лучом перебивать урон от 4 старых био-плазменных торпед +200% урона +20 качества с бустом внутри сборки +500% урона. А учитывая полную мою прокачку на тот момент, такой скорости ремонта невозможно достичь иначе кроме изменения характеристик модулей либо бустов корабля

**Comment by PavelZinchenko** [comment]
Античтит умеет находить туррель и другие корабли которых не должно быть у игрока.
Можете написать имя это читера с умноженным уроном?

**Comment by Bezbasheniy** [comment]
> Античтит умеет находить турель и другие корабли, которые не должны быть у игрока. Можете написать имя этого читера с умноженным уроном?

К сожалееию клавиатура не поддерживает некоторые символы из его никнэйма, на скриншоте он в списке первый.
![Screenshot_20240509-220041](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/169311403/e98f42dc-b0af-4c03-b241-2dc62715b854)

**Comment by imverybadatnames** [comment]
Hi, i am the #1 player in that list, i am on build 1905, what seems to be the multiplied damage you are talking about?

**Comment by Bezbasheniy** [comment]
> @BezbasheniyПривет, я игрок [№1](https://github.com/PavelZinchenko/event-horizon-frontier-builds/issues/1) в этом списке, у меня сборка 1905, о каком умноженном уроне вы говорите?
> 
> > Причем наверху сидит читер с умноженным уроном.

Привет. Я проверял возможности станций, в том числе и с полным заполнениеем зеленой зоны бустэрами а синей энэргоячейками, и выяснил что невозможно получить даже половину такого урона от луча на 3 секунде матча.
![Screenshot_20240509-181246](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/169311403/cad5962f-155c-49c7-8cc1-ac445857b1c9)

**Comment by imverybadatnames** [comment]
Hi, i am using the Thor.
This is the build of the Thor i am using.
![Screenshot_2024_0510_143020](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/164344125/606a962c-f5e1-4941-8d5f-ba79d46f4d55)


![Screenshot_2024_0510_143010](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/164344125/e91b584c-6eb9-470d-8d53-7ea5e083c50f)

**Comment by imverybadatnames** [comment]
https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/164344125/3e1b1690-df94-41bb-8f03-048f3f680a86

**Comment by Bezbasheniy** [comment]
> видео_20240510_145414.mp4

Обрати внимание на скриншот, тор к тому времени еще не успел заспавниться. Не нужно делать из меня придурка.

**Comment by Inedilarion** [comment]
> Обрати внимание на скриншот, тор к тому времени еще не успел заспавниться. Не нужно делать из меня придурка.

Спавн полностью зеркальный. В начале боя справа сверху видно хп и энергию вражеского Тора

**Comment by imverybadatnames** [comment]
> > video_20240510_145414.mp4
> 
> Pay attention to the screenshot, the torus had not yet spawned by that time. You don't have to make a fool out of me.

![IMG_20240510_152024](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/164344125/ea1e9937-ad83-41d7-9714-cc6bda15a666)
google translation can be a little wonky at times, but the thor spawned.

**Comment by Bezbasheniy** [comment]
К томуже на видео мы наблюдаем станцию "Полярная звезда" На арене же станция "Новогодняя"... В общем не вижу смысла продолжать эту безсмысленеую дискусию, тема была создана для внедрения нормальнлго античита в игру,  чтобы предотвращать появление читеров. Разбираться с каждым по отдельносьи не хватит ни времени ни сил, к томуже многие не зная всех возможностей компановки могут посчитать читера просто максимально прокачаным играком с какимито сверх имбалансными сборками и не поридать значения.

**Comment by Bezbasheniy** [comment]
> > > видео_20240510_145414.mp4
> > 
> > 
> > Обратите внимание на скриншот, тор к тому времени еще не спавнился. Не надо из меня дурака делать.
> 
> ![IMG_20240510_152024](https://private-user-images.githubusercontent.com/164344125/329512045-ea1e9937-ad83-41d7-9714-cc6bda15a666.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUzMjk2MzYsIm5iZiI6MTcxNTMyOTMzNiwicGF0aCI6Ii8xNjQzNDQxMjUvMzI5NTEyMDQ1LWVhMWU5OTM3LWFkODMtNDFkNy05NzE0LWNjNmJkYTE1YTY2Ni5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxMFQwODIyMTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NzI3YWQyZjcwM2ZlODE1MmZjYWYzYzM2ZWE2Zjk3MWRlODdhN2RlOThiZWIzNTM4MTA3OGYzMWU2NzU3ZjIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QZp2pw2muz2RRLct_m06HkI8cv0FXIkA8CmBgyjiOi8) Перевод Google временами может быть немного шатким, но Тор породил.

Переключи в автоматический выбор языка, тогда перевод будет точнее. Я писал о приложеном мною скриншоте.

**Comment by magne00** [comment]
>  simply a maximally pumped up player with some overly imbalanced builds and not pay attention to the significance.

Simply this

**Comment by Inedilarion** [comment]
Я провёл немного расчётов по скриншотам сборки и видео боя:
1) Время от момента когда луч долетает до базы до момента когда база лопается. Замерил секундомером на скорости воспроизведения 0.25, вышло 5,5 секунды, то есть меньше полутора секунд реального времени
2) ДПС луча по характеристикам сборки с бонусом к урону основного корабля +6500%: 6750 * 6.16 * 66 = 2 744 280
3) Примерный ДПС по видео: 10439430/1.5 = 6 959 620 (на самом деле ещё немного больше, потому что время действия было меньше полутора секунд)

В итоге, получилась разница в 3 раза

**Comment by Inedilarion** [comment]
> В итоге, получилась разница в 3 раза

На всякий случай дополню что это может быть вызвано тем что я где-то в своих расчётах упустил +200% урона, либо механикой действия оружия, позволяющей наносить урон втрое больше написанного в характеристиках, а не читерством

**Comment by PavelZinchenko** [comment]
Возможно там несколько лучей одновременно. Не похоже что он читер.

**Comment by magne00** [comment]
No, i can confirm not a cheat fleet. AR is just strong at the meantime. Especially maxxed out.

**Comment by Inedilarion** [comment]
На видео два выстрела Тора (судя по его шкале энергии), но к моменту уничтожения базы второй луч не успевает до неё долететь (это видно на видео). Я провёл парочку тестов, пострелял по базе из разного оружия, и выяснил что оружие с большим радиусом поражения почти всегда наносит базе двойной урон (если я верно помню, это ещё очень старый баг из ЕН РПГ, что корабли со сложной коллизией могут получать урон по площади несколько раз), возможно аннигиляционный луч имеет тот же эффект (его я проверить сейчас не могу)
![image](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/66635560/2c02020d-300b-4c15-86da-0484f57df2d6)
